### PR TITLE
Re-implement various sound effects from v0.11

### PR DESCRIPTION
**Pull request by RileyTheFox**
Reintroduces a couple of things that existed prior to the rewrite.

- Star Power reverb is back (and adjusted slightly)
- Note miss sound effect is back
- Fix some issues with stem muting and refactored the code
- Added definitions and code to play Overstrum SFX (sound files still need to be finalised and added)

Includes a YARG.Core update for the overstrum sample definitions

### CODE REVIEW

**Comment by RileyTheFox** [comment]
Requesting reviews to ensure drums + vocals function as expected and for code quality checks

**Review by EliteAsian123 — APPROVED** [review]

File: Assets/Script/Audio/Bass/BassStemChannel.cs [hunk 1]
```diff
@@ -64,8 +64,8 @@ public void Dispose()
                         }
                     };
                     _channelEndHandle = IsMixed
-                        ? BassMix.ChannelSetSync(_streamHandles.Stream, SyncFlags.End, 0, sync)
-                        : Bass.ChannelSetSync(_streamHandles.Stream, SyncFlags.End, 0, sync);
+                        ? BassMix.ChannelSetSync(StreamHandle, SyncFlags.End, 0, sync)
+                        : Bass.ChannelSetSync(StreamHandle, SyncFlags.End, 0, sync);
```
Comment (TheNathannator) [inline Assets/Script/Audio/Bass/BassStemChannel.cs:68]: I'd prefer if things were explicit with where the stream handle comes from, that's why I changed it when grouping the handles into a struct.

File: Assets/Script/Gameplay/Player/DrumsPlayer.cs [hunk 2]
```diff
@@ -161,6 +162,47 @@ protected override void UpdateVisuals(double songTime)
...
+            if (_isStemMuted == muted)
+            {
+                return;
+            }
+
+            bool anyDrumsStems = GlobalVariables.AudioManager.HasStem(SongStem.Drums) ||
+                                 GlobalVariables.AudioManager.HasStem(SongStem.Drums1) ||
+                                 GlobalVariables.AudioManager.HasStem(SongStem.Drums2) ||
+                                 GlobalVariables.AudioManager.HasStem(SongStem.Drums3) ||
+                                 GlobalVariables.AudioManager.HasStem(SongStem.Drums4);
```
Comment (TheNathannator) [inline Assets/Script/Gameplay/Player/DrumsPlayer.cs:176]: These should probably be checked for ahead of time when the song gets loaded.

**Review by TheNathannator — APPROVED** [review]
No major complaints from me! Just a few small things of note.

**Pull request merged**